languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -66,7 +66,49 @@ private void handleCall(
       args = Optional.of(host.polarListToJava(jArgs.get()));
     }
     try {
-      registerCall(attrName, args, callId, polarInstance);
+      Object instance;
+      if (polarInstance.getJSONObject("value").has("ExternalInstance")) {
+        long instanceId =
+            polarInstance
+                .getJSONObject("value")
+                .getJSONObject("ExternalInstance")
+                .getLong("instance_id");
+        instance = host.getInstance(instanceId);
```

Comment:
Can this code be folded into `host.toJava()`?

Reply:
After looking at `toJava()`, I think that this branch may be unnecessary because `toJava()` checks for `ExternalInstance` objects. Unless the structure of this event is a special case.

Reply:
I think all of these are handled by your suggestion?